_drafts/oryx/orchestration/app-feature.md

```diff
@@ -0,0 +1,126 @@
+---
+title: Application Feature
+description: Feature of the Oryx Application
+template: concept-topic-template
+---
+
+# Application feature
+
+`AppFeature` is a higher level collection of lower level primitives such as:
+
+<!-- TODO: Link to components -->
+- Components
+<!-- TODO: Link to providers -->
+- Providers
+- [Plugins](./app-plugins.md)
+<!-- TODO: Link to resources -->
+- Resources
+- [Feature options](#feature-options)
+- [Builder options](./index.md#options)
+
+Features are useful to structure and organise code and functionality into logical groups and to make them easier to reuse in different scenarios.
+
+Every Oryx package exposes such features for your application to be easily integrated.
+
+To register `AppFeature` with Oryx application use `appBuilder.withFeature()` API:
+
+```ts
+import { appBuilder, AppFeature } from '@spryker-oryx/core';
+
+const myFeature: AppFeature = {...};
+
+const app = appBuilder().withFeature(myFeature);
+```
+
+`AppFeature` is represented as an interface and you may create them as simple object literals or as classes/functions when you want them to be configurable and more flexible:
+
+```ts
+import { AppFeature } from '@spryker-oryx/core';
+
+const mySimpleFeature: AppFeature = {
+  components: [...],
+  providers: [...],
+}
+
+// or
+
+class MyConfigurableFeature implements AppFeature {
+  components;
+  providers;
+
+  constructor(config: MyConfigurableFeatureConfig) {
+    // Use `config` to customize your feature here
+    this.components = [...];
+    this.providers = [...];
+  }
+}
+```
+
+## Extending features
+
+You may also extend existing features but it will depend how the feature is defined.
+
+For features defined as object literals you may use spread operators to customize some parts of it:
+
+```ts
+const myExtendedFeature = {
+  ...mySimpleFeature,
+  components: [
+    ...mySimpleFeature.components,
+    ...[
+      /* Custom components here */
+    ],
+  ],
+};
+```
+
+For features defined as classes/function you would have to also use class or function to be able to accept configuration and pass it to the original feature:
+
+```ts
+class MyExtendedFeature extends MyConfigurableFeature {
+  constructor(config: MyConfigurableFeatureConfig) {
+    super(config);
+    this.components = [
+      ...this.components,
+      ...[
+        /* Custom components here */
+      ],
+    ];
+  }
+}
+```
+
+## Feature options
+
+If your feature requires some options from the consumer you may use `FeatureOptionsService.getFeatureOptions()` API:
```

Comment:
What is a `consumer` in this context?

Reply:
Feature consumer will usually be another dev who will use your feature in their application and may have to pass some options to it.